### PR TITLE
Fix dependencies for electron windows builds

### PR DESCRIPTION
**Pull request by anoadragon453**
Fix dependencies so Windows can build `electron` for itself.

Adds new npm scripts, that builds electron for a single platform.

Essentially we had lots of problems with building Riot on Windows. At the end of the day the solution roughly boils down to using `yarn`, updating the ancient Squirrel dependency we were using, and knowing the correct incantation under Windows batch to build things successfully.

We also hit [a bug](https://github.com/vuejs/vue-cli/issues/3407) in terser that causes some extra necessary steps to be added to the build process, but I imagine this will be fixed in a few days (but will require another version bump for whatever library we use that is relying on it).

For reference, here are the commands one needs to build Riot electron on Windows now:

```
yarn
yarn add https://matrix.org/packages/npm/olm/olm-3.0.0.tgz matrix-org/matrix-js-sdk#develop matrix-org/matrix-react-sdk#develop electron --network-timeout 100000
yarn run build:windows
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -36,6 +36,7 @@
     "build:bundle": "cross-env NODE_ENV=production webpack-cli -p --progress --bail --mode production",
     "build:bundle:dev": "webpack-cli --progress --bail --mode development",
     "build:electron": "npm run clean && npm run build && npm run install:electron && build -wml --ia32 --x64",
+    "build:electron:windows": "npm run build && build -w --ia32 --x64",
```
Comment (dbkr) [inline package.json]: If you're adding this we should probably add equivalents for linux & windows

**Comment by dbkr** [comment]
you also seem to have inherited vdh's script in your rebase somehow

**Comment by anoadragon453** [comment]
Oh, so I had.

**Comment by anoadragon453** [comment]
Added reasoning and explanation.

**Comment by jryans** [comment]
@anoadragon453 I ended up landing the dependency upgrade separately, as we needed it for Riot 1.0's release. This PR now has some conflicts... Maybe you'd like to open a fresh one to add new per-platform commands (which seem helpful)?

**Comment by anoadragon453** [comment]
Deprecated in favor of https://github.com/vector-im/riot-web/pull/9412

**Pull request closed without merging**